library_generation/cli/entry_point.py

```
@@ -0,0 +1,118 @@
+from library_generation.utils.generation_config_comparator import compare_config
+
+
+@click.group(invoke_without_command=False)
+@click.pass_context
+@click.version_option(message="%(version)s")
+def main(ctx):
+    pass
+
+
+@main.command()
+@click.option(
+    "--baseline-generation-config",
+    required=False,
+    default=None,
+    type=str,
+    help="""
+    Path to generation_config.yaml that contains the metadata about
+    library generation.
+    The googleapis commit in the configuration is the oldest commit,
+    exclusively, from which the commit message is considered.
+    """,
+)
+@click.option(
+    "--current-generation-config",
+    required=False,
+    default=None,
+    type=str,
+    help="""
+    Path to generation_config.yaml that contains the metadata about
+    library generation.
+    The googleapis commit in the configuration is the newest commit,
+    inclusively, to which the commit message is considered.
+    """,
+)
+@click.option(
+    "--repository-path",
+    type=str,
+    default=".",
+    show_default=True,
+    help="""
+    The repository path to which the generated files
+    will be sent.
+    If not specified, the repository will be generated to the current working
+    directory.
+    """,
+)
+def generate(
+    baseline_generation_config: str,
+    current_generation_config: str,
+    repository_path: str,
+):
+    default_generation_config = f"{os.getcwd()}/generation_config.yaml"
+    if baseline_generation_config is None and current_generation_config is None:
+        if not os.path.isfile(default_generation_config):
+            raise FileNotFoundError(
+                f"{default_generation_config} does not exist when "
+                f"both baseline_generation_config and current_generation_config"
+                f" are not specified."
+            )
+        current_generation_config = default_generation_config
+    elif current_generation_config is None:
```

If `current_generation_config` is none, and `baseline_generation_config` is provided, I think we should throw an error. `current_generation_config` should be the source of truth for library generation.

---

Done.